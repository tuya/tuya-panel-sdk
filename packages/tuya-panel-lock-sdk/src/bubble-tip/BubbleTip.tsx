import { memo } from 'react';
import BubbleTipBase from './BubbleTip.Base';
import ComposeHOC from './ComposeHOC';

/**
 * @desc BubbleTip 主组件的静态类型，默认是Base
 * @childType Base
 */
interface IBubbleTipStatic {
  Base: typeof BubbleTipBase;
}

const BubbleTipBaseMemo = memo(BubbleTipBase);

/**
 * @desc 默认使用的是Base的动画效果
 */
const BubbleTip: IBubbleTipStatic = props => {
  return ComposeHOC(props, BubbleTipBaseMemo);
};

BubbleTip.Base = props => {
  return ComposeHOC(props, BubbleTipBaseMemo);
};

export default BubbleTip;
