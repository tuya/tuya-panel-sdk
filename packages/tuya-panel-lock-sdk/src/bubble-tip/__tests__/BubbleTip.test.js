import React from 'react';
import { View, Text } from 'react-native';
import { shallow } from 'enzyme';
import BubbleTip from '../index';

const configList = [
  {
    text: 1,
    key: 1,
    type: 'safe',
    wrapStyle: {
      backgroundColor: 'skyblue',
    },
    textStyle: {
      color: 'red',
    },
  },
  {
    text: 2,
    key: 2,
  },
];

describe('BubbleTip components', () => {
  it('BubbleTip config render', () => {
    const wrapper = shallow(<BubbleTip configList={configList} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('BubbleTip define render', () => {
    const wrapper = shallow(
      <BubbleTip>
        {configList.map(item => {
          return (
            <View key={item.key}>
              <Text>{item.text}</Text>
            </View>
          );
        })}
      </BubbleTip>
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('BubbleTip.Base config render', () => {
    const wrapper = shallow(
      <BubbleTip.Base
        runInTime={500}
        waitingTime={800}
        leavingTime={1000}
        configList={configList}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
