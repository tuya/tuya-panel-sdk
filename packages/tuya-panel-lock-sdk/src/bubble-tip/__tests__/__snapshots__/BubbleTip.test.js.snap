// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BubbleTip components BubbleTip config render 1`] = `
<Memo(BubbleTipBase)>
  <View
    key="1"
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 20,
          "flexDirection": "row",
          "justifyContent": "center",
          "paddingHorizontal": 60,
          "paddingVertical": 20,
          "width": 360,
        },
        Object {
          "backgroundColor": "#239c8e",
        },
        Object {
          "backgroundColor": "skyblue",
        },
      ]
    }
  >
    <Image
      source={1}
      style={
        Array [
          Object {
            "height": 32,
            "marginRight": 12,
            "width": 32,
          },
          undefined,
        ]
      }
    />
    <TYText
      align={null}
      color={null}
      numberOfLines={1}
      size={null}
      style={
        Array [
          Object {
            "color": "white",
          },
          Object {
            "color": "red",
          },
        ]
      }
      text={1}
      type={null}
      weight={null}
    />
  </View>
  <View
    key="2"
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 20,
          "flexDirection": "row",
          "justifyContent": "center",
          "paddingHorizontal": 60,
          "paddingVertical": 20,
          "width": 360,
        },
        Object {
          "backgroundColor": "#239c8e",
        },
        undefined,
      ]
    }
  >
    <Image
      source={1}
      style={
        Array [
          Object {
            "height": 32,
            "marginRight": 12,
            "width": 32,
          },
          undefined,
        ]
      }
    />
    <TYText
      align={null}
      color={null}
      numberOfLines={1}
      size={null}
      style={
        Array [
          Object {
            "color": "white",
          },
          undefined,
        ]
      }
      text={2}
      type={null}
      weight={null}
    />
  </View>
</Memo(BubbleTipBase)>
`;

exports[`BubbleTip components BubbleTip define render 1`] = `
<Memo(BubbleTipBase)>
  <View
    key="1"
  >
    <Text>
      1
    </Text>
  </View>
  <View
    key="2"
  >
    <Text>
      2
    </Text>
  </View>
</Memo(BubbleTipBase)>
`;

exports[`BubbleTip components BubbleTip.Base config render 1`] = `
<Memo(BubbleTipBase)
  leavingTime={1000}
  runInTime={500}
  waitingTime={800}
>
  <View
    key="1"
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 20,
          "flexDirection": "row",
          "justifyContent": "center",
          "paddingHorizontal": 60,
          "paddingVertical": 20,
          "width": 360,
        },
        Object {
          "backgroundColor": "#239c8e",
        },
        Object {
          "backgroundColor": "skyblue",
        },
      ]
    }
  >
    <Image
      source={1}
      style={
        Array [
          Object {
            "height": 32,
            "marginRight": 12,
            "width": 32,
          },
          undefined,
        ]
      }
    />
    <TYText
      align={null}
      color={null}
      numberOfLines={1}
      size={null}
      style={
        Array [
          Object {
            "color": "white",
          },
          Object {
            "color": "red",
          },
        ]
      }
      text={1}
      type={null}
      weight={null}
    />
  </View>
  <View
    key="2"
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 20,
          "flexDirection": "row",
          "justifyContent": "center",
          "paddingHorizontal": 60,
          "paddingVertical": 20,
          "width": 360,
        },
        Object {
          "backgroundColor": "#239c8e",
        },
        undefined,
      ]
    }
  >
    <Image
      source={1}
      style={
        Array [
          Object {
            "height": 32,
            "marginRight": 12,
            "width": 32,
          },
          undefined,
        ]
      }
    />
    <TYText
      align={null}
      color={null}
      numberOfLines={1}
      size={null}
      style={
        Array [
          Object {
            "color": "white",
          },
          undefined,
        ]
      }
      text={2}
      type={null}
      weight={null}
    />
  </View>
</Memo(BubbleTipBase)>
`;
